pkg/collector/corechecks/ebpf/probe/ebpfcheck/probe.go

```diff
@@ -761,27 +760,43 @@ func hashMapNumberOfEntriesWithBatch(mp *ebpf.Map, buffers *entryCountBuffers, m
 			attr.InBatch = NewPointer(unsafe.Pointer(&buffers.cursor[0]))
 		}
 
+		// Sanity checks before we call the syscall, we don't want to overwrite memory, it causes
+		// hard to debug issues
+		if len(buffers.keys) < int(attr.Count*mp.KeySize()) {
+			return -1, fmt.Errorf("invalid keys buffer size: %d < %d, batchSize = %d", len(buffers.keys), attr.Count*mp.KeySize(), batchSize)
+		}
+		if len(buffers.values) < int(attr.Count*mp.ValueSize()) {
+			return -1, fmt.Errorf("invalid values buffer size: %d < %d ; %d", len(buffers.values), attr.Count*mp.ValueSize(), batchSize)
+		}
+		if len(buffers.cursor) < int(mp.KeySize()) {
+			return -1, fmt.Errorf("invalid cursor buffer size: %d < %d", len(buffers.cursor), mp.KeySize())
```

Comment:
I'm OK with these checks for now, but they do feel like more asserts that indicate a bug rather than any kind of runtime error.

Reply:
Yes, I wanted to leave these ones just to avoid panics and get better error messages if something fails. I can remove them later if we don't replicate the panics